tests/v2/Controller/ProController/TogglePermission.php

```diff
@@ -79,4 +79,44 @@ public function provideTestToggleBeneficiaryManagement(): ?\Generator
         yield 'Should toggle beneficiary management permission' => [MembreCentre::MANAGE_BENEFICIARIES_PERMISSION];
         yield 'Should toggle pro management permission' => [MembreCentre::MANAGE_PROS_PERMISSION];
     }
+
+    public function testCanNotGiveNotOwnedPermission(): void
+    {
+        // Test only for beneficiary management as pro list is unreachable without pro management permission
+        $testedPermission = MembreCentre::MANAGE_BENEFICIARIES_PERMISSION;
+        $ownedPermission = MembreCentre::MANAGE_PROS_PERMISSION;
+
+        self::ensureKernelShutdown();
+        $client = self::createClient();
+        $user = UserFactory::findByEmail(MemberFixture::MEMBER_MAIL_WITH_RELAYS_SHARED_WITH_MEMBER)->object();
+        $client->loginUser($user);
+
+        $relay = $user->getSubjectMembre()->getCentres()->first();
+        $userRelay = $user->getUserRelay($relay);
+
+        $userRelay->removePermission($testedPermission);
+        $this->getEntityManager()->flush();
+
+        self::assertEquals([
+            $testedPermission => false,
+            $ownedPermission => true,
+        ], $userRelay->getDroits());
+
+        // Request to pro list, with relay query param
+        $crawler = $client->request('GET', sprintf('/pro?relay=%s', $relay->getId()));
+        $ownedPermissionButtonCount = $crawler->filter(sprintf('button#%s', $ownedPermission))->count();
+        $testedPermissionButtonCount = $crawler->filter(sprintf('button#%s', $testedPermission))->count();
+
+        // Not owned permission button is not shown
+        self::assertGreaterThan(0, $ownedPermissionButtonCount);
+        self::assertEquals(0, $testedPermissionButtonCount);
+
+        // Fetch the first pro in list
+        $firstUsernameInList = $crawler->filter('td.border-0.h-100.w-25.text-start.align-middle.bold > span.text-grey')->html();
+        $firstUserInList = UserFactory::find(['username' => $firstUsernameInList])->object();
+
+        // Can not update permission
+        $client->request('GET', sprintf(self::URL, $firstUserInList->getSubjectMembre()->getId(), $relay->getId(), $testedPermission));
+        self::assertResponseStatusCodeSame(403);
```

Comment:
I think this should be a separated test. And you could use username from fixtures or db rather than the crawler, wich will be broken each time we update the UI.